src/os/state/basestatemanager.js

```diff
@@ -141,17 +141,77 @@ os.state.BaseStateManager.prototype.registerPersistable = function(key, clazz) {
 
 
 /**
- * Gets a registered {@link os.IPersistable} class.
+ * Gets an instance of a registered {@link os.IPersistable} class.
  *
  * @param {string} key The object key
  * @return {os.IPersistable}
+ * @suppress {checkTypes} To allow the catch() failover to exist
  */
 os.state.BaseStateManager.prototype.getPersistable = function(key) {
-  if (key in this.persistableMap_) {
-    return new this.persistableMap_[key]();
+  var instance = null;
+  if (this.isPersistable(key)) { // check for nulls, etc
+    var Clazz = this.persistableMap_[key];
+    try {
+      instance = new Clazz();
+    } catch (e) {
+      var args = [this.log, `Registered persistance class for "${key}"`];
```

Comment:
Nit: persistence is misspelled in every log statement as "persist*a*nce"

Reply:
Oh good, I thought I was crazy and didn't even check the spelling of the other message in here that I was using as my archetype. Updated my new ones and the old one too.